Add (#851): Solr IndexerBolt basic test / Refactor Solr tests

mvolikas opened the pull request.

jnioche commented:
Thanks @mvolikas 
StatusBoltTest.java and IndexerBoltTest.java share a lot of code, what about having them extend an abstract class where all that code would be put in?
We need the usual license headers in all files.
Is the content of _resources/cores_ the same as what is in _cores_? Instead of having duplicates, could we have only one copy or use links?

rzo1 commented on external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java:
```diff
@@ -0,0 +1,113 @@
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.*;
```
Can we avoid * imports?

rzo1 commented on external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java:
```diff
@@ -0,0 +1,113 @@
+import org.apache.stormcrawler.TestUtil;
+import org.apache.stormcrawler.indexing.AbstractIndexerBolt;
+import org.apache.stormcrawler.solr.bolt.IndexerBolt;
+import org.junit.*;
```
Can we avoid * imports?

rzo1 commented on external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java:
```diff
@@ -0,0 +1,113 @@
+package org.apache.stormcrawler.solr.persistence;
```
Requires a license header

rzo1 commented on external/solr/src/test/resources/cores/docs/conf/schema.xml:
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Requires a license header. You can just run with the `rat` profile locally, so you see all the complains ;-)

rzo1 commented on external/solr/src/test/resources/cores/docs/conf/solrconfig.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<config>
```
Header

rzo1 commented on external/solr/src/test/resources/cores/docs/core.properties:
```diff
@@ -0,0 +1,6 @@
+#Written by CorePropertiesLocator
```
Header

rzo1 commented on external/solr/src/test/resources/cores/metrics/conf/schema.xml:
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Header

rzo1 commented on external/solr/src/test/resources/cores/metrics/conf/solrconfig.xml:
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<config>
```
Header

rzo1 commented on external/solr/src/test/resources/cores/metrics/core.properties:
```diff
@@ -0,0 +1,6 @@
+#Written by CorePropertiesLocator
```
Header

rzo1 commented on external/solr/src/test/resources/cores/solr.xml:
```diff
@@ -0,0 +1,2 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Header

rzo1 commented on external/solr/src/test/resources/cores/status/core.properties:
```diff
@@ -0,0 +1,6 @@
+#Written by CorePropertiesLocator
```
Header

rzo1 reviewed (commented):
Might be worth to extract an abstract class for common code fragements.

mvolikas commented:
Thank you both for the comments! I removed the duplicate docs and added a check to ensure the test document is inside the Solr core. Please have a look at the changes and let me know if anything can be improved :)

jnioche commented on external/solr/src/test/java/org/apache/stormcrawler/solr/persistence/IndexerBoltTest.java:
```diff
@@ -0,0 +1,123 @@
+/**
+ * Licensed to DigitalPebble Ltd under one or more contributor license agreements. See the NOTICE
```
This is not the right header I'm afraid.  It should not contain any reference to DigitalPebble at all.
Please have a look at the Java classes in the main branch

mvolikas replied:
Sorry about that. I updated the headers - the headers in branch 851 seem to be the old ones.

jnioche reviewed (approved):
LGTM!

rzo1 commented:
Thx @mvolikas
